server/src/service.rs

```diff
@@ -114,16 +114,70 @@ impl Service {
     }
 
     #[zbus(out_args("collection"))]
-    pub async fn read_alias(&self, _name: &str) -> Result<ObjectPath, ServiceError> {
-        todo!()
+    pub async fn read_alias(
+        &self,
+        name: &str,
+        #[zbus(object_server)] object_server: &zbus::ObjectServer,
+    ) -> Result<OwnedObjectPath, ServiceError> {
+        let collections = self.collections.lock().await;
+
+        for collection in collections.iter() {
+            let collection_ifce_ref = object_server.interface::<_, Collection>(collection).await?;
+            let collection = collection_ifce_ref.get_mut().await;
+
+            if name == "login" {
+                if collection.label().await == name {
+                    drop(collections);
+                    return Ok(collection.path().clone());
+                }
+            } else if collection.alias().await == name {
+                drop(collections);
+                return Ok(collection.path().clone());
+            }
+        }
+
+        drop(collections);
+
+        Ok(OwnedObjectPath::default())
     }
 
     pub async fn set_alias(
         &self,
-        _name: &str,
-        _collection: ObjectPath<'_>,
+        name: &str,
+        collection: OwnedObjectPath,
+        #[zbus(object_server)] object_server: &zbus::ObjectServer,
     ) -> Result<(), ServiceError> {
-        todo!()
+        if name != "default" {
+            return Err(ServiceError::ZBus(zbus::Error::Failure(String::from(
+                "Only the 'default' alias is supported",
+            ))));
+        }
+
+        let collections = self.collections.lock().await;
+
+        for path in collections.iter() {
+            if *path == collection {
+                let collection_ifce_ref = object_server.interface::<_, Collection>(path).await?;
+                let collection = collection_ifce_ref.get_mut().await;
+
+                collection.set_alias(name).await;
+
+                tracing::info!(
+                    "Collection: {} alias updated to {}.",
+                    collection.path(),
+                    name
+                );
+                drop(collections);
+
+                return Ok(());
+            }
+        }
+
+        drop(collections);
+
+        Err(ServiceError::NoSuchObject(String::from(
+            "The collection does not exist",
```

Comment:
Which collection? add the alias to the error message

Reply:
Done, thanks!